### DISCUSSION TITLE
Should starship work in an ssh shell?

### DISCUSSION THREAD

**Discussion by pnunn** [starship/starship#3369] [Q&A] [answered]
I'm just starting out with starship and its really good so far. Using a fish shell.

I have it installed and working on my laptop and on one of my servers in the server console and it looks fine.

However, when I ssh into the server instead of logging into the console, I have no prompt at all. it is completely blank.

Am I missing something here?

Ta Peter.

**Comment by chipbuster** [comment]
Do you still have a shell? Like you can run commands, but there's no prompt at all?

**Comment by pnunn** [comment]
Yes, the shell still works. If I remove the starship initialisation from config.fish the fish prompt comes back (which is also funky on the ssh consoles I might add).

**Reply by chipbuster** [reply]
That's very strange. When you say the default fish prompt is funky, what do you mean? Could you provide a screenshot if it helps demonstrate the strangeness?

**Comment by pnunn** [comment]
Sure can.. :)  

Here I am logged into the docker host with starship enabled.

![image](https://user-images.githubusercontent.com/1034315/147839537-e799e5aa-2865-4407-b805-1a8906067863.png)

Same machine, logged into the console

![image](https://user-images.githubusercontent.com/1034315/147839564-c7db747f-2f15-41ef-9f55-78f13edd43d4.png)

SSH with starship disabled

![image](https://user-images.githubusercontent.com/1034315/147839581-fa1f8d59-aff8-4e44-9e54-4921f7063514.png)

Hope it shows something :)  Thanks.

**Reply by chipbuster** [reply]
Hm. That almost looks like locale isn't being set correctly. When you're on SSH, what is the output you get with the following commands?

```
echo $LANG
echo $LC_ALL
echo $TERM
```

**Comment by pnunn** [comment]
echo $LANG en_US.UTF-8
echo $LC_ALL

 echo $TERM xterm-256color

Which is the same as the laptop that is working.

**Reply by chipbuster** [reply]
Huh, that's strange.

What about the following commands?

echo -e "\xf0\x9f\x90\x8d"
echo -e "\xee\x82\xa0"

**Reply by chipbuster** [reply]
Also, I notice that your terminal seems to be having trouble understanding color-change codes (there are `?` marks every time the color changes). Can you see what the following two snippets look like for you? I've attached what I see when I try it with fish.

```
for color in (set_color --print-colors)
  set_color $color; echo "TEST TEXT"
end
```

![image](https://user-images.githubusercontent.com/4605384/147840219-7c419ed1-059d-4c94-b43d-991d525b11dd.png)


```
for color in (set_color --print-colors)
  set_color -b $color; echo "TEST TEXT"
end
```

![image](https://user-images.githubusercontent.com/4605384/147840223-73d30923-d7ab-4f64-9015-793b6e2cb62f.png)

**Reply by pnunn** [reply]
![image](https://user-images.githubusercontent.com/1034315/147840484-aa30473c-2445-4836-9e1e-dd0bf08fba08.png)

Seems to work OK??  I wondered what the ?'s were, that's why I started chasing a different prompt. Some of the fish prompts don't work at all so maybe it is something with the terminal. I'm using KDE NEON and  Yakake on the laptop. The ssh sessions are running tmux if that makes a difference.

**Reply by pnunn** [reply]
I've also just put an intentional error into the toml file to see if it is even loading. It produced an error on starting the ssh session, so it seems starship is infact running, just not dispalying anything?? Sigh.

**Comment by pnunn** [comment]
Just had a thought that it migth be an issue with the toml file, but renaming that and restarting starship results in the same thing unfortunately.

**Reply by pnunn** [reply]
For what its worth, my .tmux.conf and config.fish from the remote machine.
[configs.zip](https://github.com/starship/starship/files/7797367/configs.zip)

**Comment by pnunn** [comment]
Symbols are displaying OK...

![image](https://user-images.githubusercontent.com/1034315/147840118-18896e0f-ac8e-466c-9484-03d9a823fbb4.png)

**Comment by pnunn** [comment, marked as answer]
Right.. its an issue with fish.

I just enabled starship in bash, then started a bash shell from fish (in the ssh session) and boom... starship works fine.

I'll start playing with my fish config.

**Comment by pnunn** [comment]
Anyone have any clues with this? I'd rather not give up on fish, but I guess I could go back to zsh.

Ta.. Peter

**Reply by chipbuster** [reply]
One thought that occurs is that it's possible that the system bashrc (or /etc/profile) is setting something that fish isn't.

To check this, you would probably need to set your shell to bash (or zsh), log in via SSH, then run `export` and save the results, switch to fish, and do the same thing.

Post the two here and we'll see what we can find.

**Reply by pnunn** [reply]
Thanks @chipbuster for the input again.. Happy new year to you!!

I have the files here... interestingly if I run fish after the initial login shell is bash (so run fish after logging in) the prompt works fine (as does the fish prompt??) Something is up here :). That file is bash-fish.export.

I hope they make sense
[exports.zip](https://github.com/starship/starship/files/7811271/exports.zip)
.

**Reply by chipbuster** [reply]
Sorry for the delay.

I do see something interesting here, though it's not what I would expect: your locale in bash and fish-after-bash is `C.UTF-8`, while in plain fish, it's set to `en_US.UTF-8`.

What if you set `LANG` to `C.UTF-8`. Does that change anything?

**Reply by pnunn** [reply]
Sorry for the huge delay. I got distracted with the Linux Australia Conference.  I've found the problem thanks to your thoughts. The issue is that none of the machines had a default locale set. I installed en_AU.UTF-8 on all of them, and then set the LANG and LC_ALL to be that locale, and not only does spaceship now work, but the fish prompts are correct as well..

Thank you so much. Now to do another ansible script to install spaceship on all of the machines.

Peter.
